modules/pricing/state_and_plan_content_from_json.tf

```
@@ -55,7 +55,7 @@ locals {
 
   instances_map = local.input_type == "plan" ? local.normalized_plan_instances : local.normalized_state_instances
 
-  instances_stripped_keys = local.input_type == "empty" ? [] : concat([
+  instances_stripped_keys = local.instances_map == {} ? [] : concat([
```

```suggestion
  instances_stripped_keys = local.input_type == "empty" || length(local.instances_map) == 0 ? [] : concat([
```

I think this is a safer option. Do you agree?